api/proxy/proxy.go

```diff
@@ -405,3 +427,33 @@ func (p *proxy) applyDefaultForNonDAVRequest(val api.DefaultNonDAVRequest, w htt
 		return
 	}
 }
+
+func (p *proxy) validClient(userAgent string) bool {
+
+	match := p.VersionRegex.FindStringSubmatch(userAgent)
```

Comment:
Use FindString instead of FindStringSubmatch and do

```
match := p.VersionRegex.FindString(userAgent)

if match != "" { // we have a match
    // check version is allowed
}

// we always let them trough here
```

Reply:
With what I did, I already get an array with `[major, minor, build]`, so I can already compare without doing any other manipulation (which I would with what you suggest, right?).

Reply:
Yours is just difficult to read and needed to go for the documentation twice :)

If you are comfortable with that, leave it, but make sure you get some set of user agents used by our users and curl-them against your logic.

Reply:
Yes, it's harder, but I did it to try to be a bit more efficient. I'll add some comments to be easier.
I did some tests and it seems to be working well.
(but enjoy your holidays.. we will have time to check this when you return, probably)

Reply:
probably not so I merge now :)